scripts/Activity.vue

```diff
@@ -61,19 +61,20 @@ async function update(fNewOnly = false, sync = true) {
 
     // Prevent the user from spamming refreshes
     if (cNet.historySyncing) return;
-
-    updating.value = true;
     let arrTXs;
-
-    if (txCount !== cNet.arrTxHistory.length || !sync) {
-        arrTXs = cNet.arrTxHistory;
-    } else {
-        arrTXs = await cNet.syncTxHistoryChunk(fNewOnly);
+    try {
+	updating.value = true;
+	
+	if (txCount !== cNet.arrTxHistory.length || !sync) {
+            arrTXs = cNet.arrTxHistory;
+	} else {
+            arrTXs = await cNet.syncTxHistoryChunk(fNewOnly);
+	}
+	txCount = arrTXs.length;
+	if (!arrTXs || cNet !== getNetwork()) return;
+    } finally {
+	updating.value = false;
```

Comment:
I think this section should either be commented a bit, or cleaned somehow, it's a bit 'messy' looking and takes a moment to unwrap.

Reply:
Done